Assets/com.gdcbd.bossbattle/Scripts/Managers/ObjectPoolManager.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections.Generic;
+using com.gdcbd.bossbattle.utility;
+using UnityEngine;
+
+namespace com.gdcbd.bossbattle
+{
+    public class ObjectPoolManager : PersistentMonoSingleton<ObjectPoolManager>
```

Comment:
Use unity's native object pooling instead of writing your own

Reply:
native object pool for projectile added